src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1620,6 +1620,31 @@ namespace cryptonote
     return m_mempool.check_for_key_images(key_im, spent);
   }
   //-----------------------------------------------------------------------------------------------
+  uint64_t core::get_total_burnt_loki(const uint64_t start_offset, const size_t count)
+  {
+    uint64_t burnt_loki = 0;
+    if (count)
+    {
+      const uint64_t end = start_offset + count - 1;
+      m_blockchain_storage.for_blocks_range(start_offset, end,
+        [this, &burnt_loki](uint64_t, const crypto::hash& hash, const block& b){
+      std::vector<transaction> txs;
+      std::vector<crypto::hash> missed_txs;
+      uint64_t coinbase_amount = get_outs_money_amount(b.miner_tx);
+      this->get_transactions(b.tx_hashes, txs, missed_txs);      
+      for(const auto& tx: txs)
+      {
+        if(b.major_version >= HF_VERSION_FEE_BURNING)
```

Comment:
This check is better done on the outside, so you can skip batches of TX that aren't relevant or better yet ensuring that start offset starts at HF_VERSION_FEE_BURNING to skip a lot of needlessly churning blocks.

But, on closer look I'd suggest accumulating the burn in `get_coinbase_tx_sum`, instead of returning a pair, return a struct with 3 fields, {total_fee_amount, emission_amount, total_burnt} because these values are ecommonly going to be used in relation to each other, removing the need to loop twice over the same set of data to get the equivalent result.

I realise I suggested making something akin to get_coinbase_tx_sum but it seems better off calculating it in get_coinbase_tx_sum in the end.

Reply:
Thanks for the review. I've updated get_coinbase_tx_sum to return a tuple with the additional int for burnt-loki